Mathlib/MeasureTheory/Function/L1Space.lean

```diff
@@ -237,18 +237,20 @@ theorem hasFiniteIntegral_norm_iff (f : α → β) :
 
 theorem hasFiniteIntegral_toReal_of_lintegral_ne_top {f : α → ℝ≥0∞} (hf : ∫⁻ x, f x ∂μ ≠ ∞) :
     HasFiniteIntegral (fun x ↦ (f x).toReal) μ := by
-  have h x : ‖(f x).toReal‖ₑ = .ofNNReal ⟨(f x).toReal, ENNReal.toReal_nonneg⟩ := by
-    rw [Real.enorm_of_nonneg]
+  have h x : ‖(f x).toReal‖ₑ = .ofReal (f x).toReal := by
+    rw [Real.enorm_of_nonneg ENNReal.toReal_nonneg]
   simp_rw [hasFiniteIntegral_iff_enorm, h]
   refine lt_of_le_of_lt (lintegral_mono fun x => ?_) (lt_top_iff_ne_top.2 hf)
   by_cases hfx : f x = ∞
   · simp [hfx]
   · lift f x to ℝ≥0 using hfx with fx h
     simp [← h, ← NNReal.coe_le_coe]
 
-lemma hasFiniteIntegral_toReal_iff {f : α → ℝ≥0∞} (hf_ne_top : ∀ᵐ x ∂μ, f x ≠ ∞) :
+lemma hasFiniteIntegral_toReal_iff {f : α → ℝ≥0∞} (hf : ∀ᵐ x ∂μ, f x ≠ ∞) :
     HasFiniteIntegral (fun x ↦ (f x).toReal) μ ↔ ∫⁻ x, f x ∂μ ≠ ∞ := by
-  simp [hasFiniteIntegral_iff_enorm, Real.enorm_of_nonneg ENNReal.toReal_nonneg]
+  have : ∀ᵐ x ∂μ, .ofReal (f x).toReal = f x := by filter_upwards [hf] with x hx; simp [hx]
```

Comment:
Should this be a separate lemma?

Reply:
Not such what you mean. This have is just an ae version of `ENNReal.ofReal_toReal`